controllers/hyperconverged/hyperconverged_controller.go

```diff
@@ -355,7 +357,10 @@ func (r *ReconcileHyperConverged) doReconcile(req *common.HcoRequest) (reconcile
 		// get into upgrade mode
 
 		r.upgradeMode = true
-
+		err := validateUpgradePatches(req)
+		if err != nil {
+			return reconcile.Result{Requeue: true}, err
```

Comment:
Maybe I'm wrong here, but returning requeue=true  won't solve the problem. The error will happen if we can't open the file, if its format is wrong. Another option is when the HyperConverged version is with a wrong format.

For the first cases WDYT about validating on boot, and fail the whole process? that way we could make sure we're releasing a valid file, because no test will run.

For the last one - maybe we can check the version in the create/update webhooks?

Reply:
If the json file is corrupted/wrong, requeue=true will not fix it for us.
But for sure the upgrade lanes will detect it so the risk of releasing with a broken file is really low.
On the other side, the earlier we fail the better it is.

Reply:
Still, returning requeue=true here may cause an endless loop - because nothing can fix these problems.